homeassistant/components/tessie/entity.py

```diff
@@ -0,0 +1,47 @@
+"""Tessie parent entity class."""
+
+
+from typing import Any
+
+from aiohttp import ClientSession
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MODELS
+from .coordinator import TessieDataUpdateCoordinator
+
+
+class TessieEntity(CoordinatorEntity[TessieDataUpdateCoordinator]):
+    """Parent class for Tessie Entities."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: TessieDataUpdateCoordinator,
+        key: str,
+    ) -> None:
+        """Initialize common aspects of a Tessie entity."""
+        super().__init__(coordinator)
+        self.vin: str = coordinator.vin
+        self.key: str = key
+        self.session: ClientSession = coordinator.session
```

Comment:
This looks unused. Ideally we don't want this level of implementations details such as the client session exposed to the entity

Reply:
The library requires the client session in every API call (this becomes more apparent when you make changes). I agree its not ideal, but the library doesnt have a init class where this would normally get set once.

Reply:
I don't see it used anywhere in the code in this PR?

Reply:
Yeah ok, foreshadowing. I have removed it.
It will come back in the next platform PR.

Reply:
Sounds good. We can discuss solutions to avoid it than if its possible when there is more context